src/ai/backend/manager/sweeper/kernel.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+import logging
+from collections import defaultdict
+from contextlib import asynccontextmanager as actxmgr
+from contextlib import suppress
+from typing import AsyncIterator, override
+
+import aiotools
+import sqlalchemy as sa
+from sqlalchemy.orm import load_only, noload
+
+from ai.backend.common.events import KernelLifecycleEventReason
+from ai.backend.common.validators import TimeDelta
+from ai.backend.logging import BraceStyleAdapter
+
+from ..api.context import RootContext
+from ..config import session_hang_tolerance_iv
+from ..models import DEAD_KERNEL_STATUSES, DEAD_SESSION_STATUSES, KernelRow, SessionRow
+from .base import AbstractSweeper
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+
+class KernelSweeper(AbstractSweeper):
+    _root_ctx: RootContext
```

Comment:
You should only get the values **you need**.

Reply:
Resolved via dc7bbf2ff61a8e7737ca9f8288c0a9d887670e33